POST endpoint and validation added

SlyCright opened the pull request.

SlyCright committed:
Validation added

philkuzmin commented on src/main/java/educationalproject/programmingstuff/controller/UserController.java:
```diff
@@ -1,38 +1,28 @@
-    private final OrderRepository orderRepository;
-
-    private final TestService testService;
+    private final educationalproject.programmingstuff.servicies.TestService testService;
```
import that shit

philkuzmin commented on src/main/java/educationalproject/programmingstuff/controller/UserController.java:
```diff
@@ -1,38 +1,28 @@
 
 @RestController
 @RequiredArgsConstructor
+@RequestMapping("/users")
```
@Validated

philkuzmin commented on src/main/java/educationalproject/programmingstuff/controller/UserController.java:
```diff
@@ -48,7 +38,11 @@
         return ResponseEntity.ok(users);
+    }
 
+    @PutMapping
```
@PostMapping

philkuzmin commented on src/main/java/educationalproject/programmingstuff/controller/UserController.java:
```diff
@@ -48,7 +38,11 @@
+    }
 
+    @PutMapping
+    public ResponseEntity<UserResponseDto> postUser(@RequestBody UserResponseDto user) { //todo q: optional in swagger
```
create UserCreateRequestDto

philkuzmin commented on src/main/java/educationalproject/programmingstuff/servicies/UserService.java:
```diff
@@ -0,0 +1,14 @@
+
+    List<UserResponseDto> getAllUsers();
+
+    UserResponseDto saveUser(UserResponseDto user);
```
createUser

philkuzmin commented on src/main/java/educationalproject/programmingstuff/servicies/UserServiceImpl.java:
```diff
@@ -0,0 +1,47 @@
+    public UserResponseDto saveUser(UserResponseDto userResponseDto) {
+
+        User userEntity = userResponseMapper.makeUserEntityOf(userResponseDto);
+        userRepository.saveAndFlush(userEntity);
```
check if save() returns created entity

SlyCright committed:
Bug fixes for PR#5:
  - UserCreateRequestDto added
  - UserCreateRequestMapper added
  - Minor bug fixing
